src/Lean/Meta/Basic.lean

```diff
@@ -1007,7 +1016,7 @@ mutual
 
   private partial def isClassExpensive? (type : Expr) : MetaM (Option Name) :=
     withReducible do -- when testing whether a type is a type class, we only unfold reducible constants.
-      forallTelescopeReducingAux type none fun _ type => do
+      forallTelescopeReducingAux (← instantiateMVars type) none fun _ type => do
```

Comment:
Is this even necessary?  There's a `whnf` below.

Reply:
I do not remember whether this change was necessary on this test in particular. 
I think it is a problem when we call this and the argument `type` of `isClassExpensive?` is itself a metavariable, then we might not have telescoped inside all pi-types.
I will investigate, and either add a test where this change is necessary (or confirm it's necessary on the already added test) or I will revert this. (this will take a few days, since I only have built Lean from source on my office computer.)

Reply:
It is necessary, even for this test. I moved it 2 lines lower to a (hopefully) more intuitive location. The reason why it is needed is to make the `type.getAppFn` work: we should instantiate enough metavariables to make sure we get the correct `type.getAppFn`.